### PR TITLE
AssignRoleToUsers should not be implied by EditUsers

### LINKED ISSUE #16756: AssignRoleToUsers should not be implied by EditUsers
In the permission structure we currently have. The `AssignRoleToUsers` permission should NOT be implied by `EditUser`. `AssignRoleToUsers` should be explicitly granted instead.

Change

https://github.com/OrchardCMS/OrchardCore/blob/dd03cf8c998b014303728e20b398239e49ff3dc8/src/OrchardCore/OrchardCore.Users.Core/CommonPermissions.cs#L24

to 

```
    public static readonly Permission AssignRoleToUsers = new("AssignRoleToUsers", "Assign any role to users", true);
```

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #16756

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
@hishamco the issue is tagged with 3.0. It is probably better to work on these issues when we are getting ready to release 3.0 to avoid conflicts and extra work

**Comment by hishamco** [comment]
To be honest I'm not sure what is the roadmap, I think we should make this clear to everyone on the team

**Comment by MikeAlhayek** [comment]
The issue is assigned to 3.0 milestones. That means it won't be merged in 2.x.

**Comment by hishamco** [comment]
I changed it to `2.1.0`

**Comment by MikeAlhayek** [comment]
You can't change it. It's a breaking change. That's why it was tagged with 3.0.

**Comment by MikeAlhayek** [comment]
its not a bug and has been like this for a long time. But we should not need that EditUsers permissions there.

**Comment by MikeAlhayek** [comment]
@hishamco please document this change into 3.0.0 release notes and we'll merge it.

File: src/docs/releases/3.0.0.md [hunk 1]
```diff
@@ -48,3 +48,7 @@ The following obsolete settings were removed from `LoginSettings` class
 #### Login View Update
 
 The `ExternalLogin` action has been removed from the `Account` controller. If you are using a custom `Login.cshtml` view or `Login` template, please update the external login form action. As of this update, the `ExternalLogin` action has been relocated to the `ExternalAuthentications` controller.
+
+#### AssignRoleToUsers Permission Update
+
+The `AssignRoleToUsers` permission should not be implied by `EditUsers`.
```
Comment (MikeAlhayek) [inline src/docs/releases/3.0.0.md:54]:
```suggestion
The `AssignRoleToUsers` permission is no longer implicitly granted by `EditUsers`. To maintain the same behavior, make sure to explicitly assign the `AssignRoleToUsers` permission to any role that already has the `EditUsers` permission.
```

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]
Please use descriptive docs in the release notes to help users understand what happened and what they should do to maintain behavior.

**Pull request merged**